### DISCUSSION TITLE
Is React Native 0.65 required to update to RNIAP `12.x`?

### DISCUSSION THREAD

**Discussion by dustinbahr** [hyochan/react-native-iap#2272] [Q&A] [answered]
[The docs](https://react-native-iap.dooboolab.com/docs/get-started#requirements) state the requirement of `react-native >= 0.65.1`.  But, I am wondering if this is truly necessary?

My app is currently running `react-native-iap` `7.5.6`, and `react-native` `0.63.5`.

I am looking at possibly upgrading to the latest `react-native-iap` `12.x`.

However, I am concerned about updating `react-native` just because of all the other modules and code in my project.

Do I really need to update RN in order to update RNIAP?

**Comment by andresesfm** [comment]
Why don't you give it a try? I can't think of a specific reason for it to be the minimum. Let us know of your findings and we can update the docs.

**Comment by dustinbahr** [comment, marked as answer]
I have tried it out, only upgrading `react-native-iap`.

So far I have not uncovered any issues that stem from the `react-native` version. Although, I don't know what I don't know...

I have upgraded to `12.x` and while the migration guides in the docs were helpful for versions `10`, `11`, and `12`, there wasn't one for the largest change from `8 -> 9` which had many undocumented breaking changes.

At this point, probably not worth going back to document those, but perhaps at least mention somewhere that if upgrading from a version less than 9, it might be best to start from scratch.

**Comment by dustinbahr** [comment]
I should also add that, after upgrading, we started experiencing long delays between requesting a subscription product purchase and when the iOS purchase flow would appear. So, we ended up staying on the old version of `react-native-iap`.
